tests/testthat/test_normalizeresiduals.R

```diff
@@ -0,0 +1,174 @@
+              expect_equal(length(tfuns2$cdf), length(tfuns2$quant))
+
+              expect_equal(length(tfuns$cdf), ncol(residT))
+
+
+              testdata <- t(as.matrix(runif(length(tfuns$cdf), 0.1, 6)))
+
+
+              get.quans <- function(index){
+                  tfuns$cdf[[index]](testdata[,index])
+              }
+
+              q <- sapply(1:ncol(testdata), get.quans)
+
+
+              get.quans2 <- function(index){
+                  tfuns2$cdf[[index]](testdata[,index])
+              }
+
+              q2 <- sapply(1:ncol(testdata), get.quans2)
+
+              expect_equal(q, q2)
+
+          })
+
+test_that('Emulator is calling characterize.emp.dist correctly for P residuals',
+          {
+              expect_equal(length(pfuns$cdf)  , length(pfuns2$cdf))
+              expect_equal(length(pfuns$quant), length(pfuns2$quant))
+
+              expect_equal(length(pfuns$cdf),  length(pfuns$quant))
+              expect_equal(length(pfuns2$cdf), length(pfuns2$quant))
+
+              expect_equal(length(pfuns$cdf), ncol(residP))
+
+
+              testdata <- t(as.matrix(runif(length(pfuns$cdf), 0.1, 6)))
+
+
+              get.quans <- function(index){
+                  pfuns$cdf[[index]](testdata[,index])
+              }
+
+              q <- sapply(1:ncol(testdata), get.quans)
+
+
+              get.quans2 <- function(index){
+                  pfuns2$cdf[[index]](testdata[,index])
+              }
+
+              q2 <- sapply(1:ncol(testdata), get.quans2)
+
+              expect_equal(q, q2)
+
+          })
+
+
+
+
+test_that('T residuals are preserved through normalize and unnormalize',
+          {
+              # convert the T residuals from their native distribution in each
+              # grid cell to N(0,1)
+              t1 <- normalize.resids(inputresids = residT, empiricalcdf = tfuns$cdf)$rn
+
+              # Convert T residuals in N(0,1) back to the native distribution
+              # in each grid cell
+              t2 <- unnormalize.resids(empiricalquant = tfuns$quant, rn = t1)$rnew
+
+              # compare t2 with the original residuals
+              maxdiff <- max(abs(t2-residT))
+              ftol <- 1.0e-6 * min(abs(residT))
+              expect_lte(maxdiff, ftol)
+          })
+
+
+test_that('P residuals are preserved through normalize and unnormalize',
+          {
+              # convert the T residuals from their native distribution in each
+              # grid cell to N(0,1)
+              p1 <- normalize.resids(inputresids = residP, empiricalcdf = pfuns$cdf)$rn
+
+              # Convert T residuals in N(0,1) back to the native distribution
+              # in each grid cell
+              p2 <- unnormalize.resids(empiricalquant = pfuns$quant, rn = p1)$rnew
+
+              # compare p2 with the original residuals
+              maxdiff <- max(abs(p2-residP))
+              ftol <- 1.0e-5 * min(abs(residP))
+              expect_lte(maxdiff, ftol)
```

Comment:
If it helps in the future, `expect_equal` and `expect_equivalent` both take a `tolerance` argument.